doc/reqparse

```diff
@@ -0,0 +1,153 @@
+                            "expected markdown or next req header marker")
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+            elif l.strip() == "-"*80:
+                if state == State.BEGIN:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected group header marker")
+                elif state == State.GROUP_HEADER:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected yaml or group header end marker")
+                elif state == State.GROUP_TEXT:
+                    group = current_entity
+                    current_entity = None
+                    state = State.REQ_HEADER_PRE
+                elif state == State.REQ_HEADER_PRE:
+                    current_entity = Entity()
+                    state = State.REQ_HEADER
+                    header_line = i
+                elif state == State.REQ_HEADER:
+                    error(filename, i, 1, "unexpected req header marker,\n"
+                            "expected markdown or req header end marker")
+                elif state == State.REQ_TEXT:
+                    reqs.append(current_entity)
+                    current_entity = None
+                    state = State.REQ_HEADER_PRE
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+            else:
+                if state == State.BEGIN:
+                    error(filename, i, 1, "unexpected character, expected group header")
+                elif state == State.GROUP_HEADER:
+                    current_entity.header_text += l
+                elif state == State.GROUP_TEXT:
+                    current_entity.text += l
+                elif state == State.REQ_HEADER_PRE:
+                    error(filename, i, 1, "expected another line of req header marker")
+                elif state == State.REQ_HEADER:
+                    current_entity.header_text += l
+                elif state == State.REQ_TEXT:
+                    current_entity.text += l
+                else:
+                    error(filename, i, 1, "something went terribly wrong")
+    if state == State.REQ_TEXT:
+            reqs.append(current_entity)
+    return (group, reqs)
+
+
+parser = argparse.ArgumentParser(prog="reqparse")
+parser.add_argument(
+    "-o", "--output", action="store",
+    help="Output file"
+)
+parser.add_argument(
+    "-f", "--format", action="store",
+    choices=["html", "markdown"], default="markdown",
+    help="Output format {html|markdown}",
+)
+
+args = parser.parse_args(sys.argv[1:])
+
+buf = ""
+with io.StringIO() as output:
+    for filename in glob.glob("requirements/*"):
+        group, reqs = parse_file(filename)
+        output.write(f"# {group.header['group']}\n")
+        output.write(group.text)
+        for r in reqs:
+            output.write(f"## {r.header['title']}\n")
```

Comment:
`###` instead of `##` would increase html's readability

Reply:
It's matter of CSS formatting.